test/specs/notification_spec.rb

```diff
@@ -2,9 +2,18 @@
 require 'pry'
 
 class Jackal::Slack::Notification
-  def post_to_slack(description, attachments)
-    $attachments = attachments
-    $description = description
+  alias_method :og_execute, :execute
```

Comment:
Instead of aliasing, just check the contents of the payload. You can check the complete array to see if the slack notification is included (`payload[:complete]`)